tests/Feature/Controller/Student/AddStudentLanguageControllerTest.php

```diff
@@ -0,0 +1,138 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tests\Feature\Controller\Student;
+
+use Tests\TestCase;
+use App\Models\Student;
+use App\Models\Resume;
+use App\Models\Language;
+use App\Http\Controllers\api\Student\AddStudentLanguageController;
+use App\Service\Student\AddStudentLanguageService;
+use Illuminate\Foundation\Testing\DatabaseTransactions;
+
+
+class AddStudentLanguageControllerTest extends TestCase
+{
+    use DatabaseTransactions;
+
+    protected $student;    
+    protected $resume;
+    protected $language;
+
+    protected function setUp(): void
+    {
+        parent::setUp();       
+
+        $this->student = Student::factory()->create();
+        
+        $this->resume = Resume::factory()->create(['student_id' => $this->student->id]);
+
+        //Language has no factory because it has fixed values in the seeder      
+        $this->language = Language::first();    
+    }
+    public function testAddStudentLanguageControllerAddsLanguageSuccessfully(): void
+    {
+        $response = $this->postJson(route('student.addlanguage', ['studentId' => $this->student->id]), [
+            'language_id' => $this->language->id,
+        ]);
+
+        $response->assertStatus(200);
+
+        $response->assertJson(['message' => 'L\'idioma s\'ha afegit']);
+    }
+
+    public function testAddStudentLanguageControllerReturns404ForInvalidStudentUuid(): void
+    {
+        $invalidStudentId = 'invalidStudentId';
+
+        $response = $this->postJson(route('student.addlanguage', ['studentId' => $invalidStudentId]), [
+            'language_id' => $this->language->id,
+        ]);
+
+        $response->assertStatus(404);
+
+        $response->assertJson(['message' => 'No s\'ha trobat cap estudiant amb aquest ID: ' . $invalidStudentId]);
+    }
+
+    public function testAddStudentLanguageControllerReturns404ForValidStudentUuidWithoutResume(): void
+    {
+        $this->resume->delete();
+
+        $response = $this->postJson(route('student.addlanguage', ['studentId' => $this->student->id]), [
+            'language_id' => $this->language->id,
+        ]);
+
+        $response->assertStatus(404);
+
+        $response->assertJson(['message' => 'No s\'ha trobat cap currículum per a l\'estudiant amb id: ' . $this->student->id]);
+    }
+
+    public function testAddStudentLanguageControllerReturns422ForNonExistentLanguageUuid(): void
+    {
+        $noExistentLanguageId = 'ab9bb2ed-8bb5-4a3a-bdb2-09cd00000f0b';
```

Comment:
Extract this as a constant. In example:

self::NON_EXISTENT_LANGUAGE_ID

Reply:
done for INVALID_STUDENT_ID, NON_EXISTENT_LANGUAGE_ID and INVALID_LANGUAGE_ID